starlite/storage/file_backend.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import annotations
+
+import pickle
+import shutil
+from datetime import datetime, timedelta
+from typing import TYPE_CHECKING, NamedTuple
+
+from anyio import Lock, Path
+from anyio.to_thread import run_sync
+
+from .base import StorageBackend
+
+if TYPE_CHECKING:
+    from os import PathLike
+
+
+class FileStorageMetadataWrapper(NamedTuple):
+    """Metadata of a session file."""
+
+    expires: datetime | None
+    data: bytes
+
+
+class FileStorageBackend(StorageBackend):
+    """Session backend to store data in files."""
+
+    __slots__ = ("path", "_lock")
+
+    def __init__(self, path: PathLike, key_prefix: str | None = None) -> None:
+        super().__init__(key_prefix=key_prefix)
+        self._lock = Lock()
+        self.path = Path(path)
+
+    def with_key_prefix(self, key_prefix: str) -> FileStorageBackend:
+        new = type(self)(path=self.path, key_prefix=key_prefix)
+        new._lock = self._lock
+        return new
+
+    @staticmethod
+    def _is_expired(wrapped_data: FileStorageMetadataWrapper) -> bool:
+        return wrapped_data.expires is not None and wrapped_data.expires > datetime.now()
+
+    @staticmethod
+    async def _load_from_path(path: Path) -> FileStorageMetadataWrapper:
+        data = await path.read_bytes()
+        wrapped_data: FileStorageMetadataWrapper = pickle.loads(data)  # nosec # noqa: SCS113
+        return wrapped_data
+
+    async def get(self, key: str) -> bytes | None:
+        path = self.path / self.make_key(key)
+        if not await path.exists():
+            return None
+
+        async with self._lock:
+            wrapped_data = await self._load_from_path(path)
+            if self._is_expired(wrapped_data):
+                await path.unlink()
+
+        return wrapped_data.data
+
+    async def set(self, key: str, value: bytes, expires: int | None = None) -> None:
+        await self.path.mkdir(exist_ok=True)
+        path = self.path / self.make_key(key)
+        wrapped_data = FileStorageMetadataWrapper(
```

Comment:
```suggestion

        wrapped_data = FileStorageMetadataWrapper(
```

Add spacings to make the code more readable